### PR TITLE
Sanitize MPI SQL queries

### LINKED ISSUE #507: Fix record linkage issues causing e2e test failures

### PR DESCRIPTION
**Pull request by DanPaseltiner**
# PULL REQUEST

## Summary
This PR introduces proper sanitization of inputs when constructing SQL queries against the MPI. This should resolve the error the record linkage service was producing during end to end tests in the starter kit and secure the record linkage service against SQL injection attacks.

## Related Issue
Fixes #507

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: tests/linkage/test_postgres_mpi_connector.py [hunk 1]
```diff
@@ -21,15 +21,25 @@ def test_generate_block_query():
...
-
-    generated_query = postgres_client._generate_block_query(block_vals)
 
-    assert expected_query_line_0 == generated_query.split("\n")[0]
-    assert expected_query_last_line == generated_query.split("\n")[-1].strip()
+    generated_query, generated_data = postgres_client._generate_block_query(block_vals)
+    assert (
+        generated_query.__str__()
+        == "Composed([SQL('SELECT patient_id, person_id, jsonb_path_query_array(patient_resource,%s) as address, jsonb_path_query_array(patient_resource,%s) as birthdate, jsonb_path_query_array(patient_resource,%s) as city, jsonb_path_query_array(patient_resource,%s) as first_name, jsonb_path_query_array(patient_resource,%s) as last_name, jsonb_path_query_array(patient_resource,%s) as mrn, jsonb_path_query_array(patient_resource,%s) as sex, jsonb_path_query_array(patient_resource,%s) as state, jsonb_path_query_array(patient_resource,%s) as zip FROM '), Identifier('test_patient_mpi'), SQL(\" WHERE CAST(jsonb_path_query_array(patient_resource, %s) as VARCHAR)= '[true]' AND CAST(jsonb_path_query_array(patient_resource, %s) as VARCHAR)= '[true]';\")])"
+    )
```
Comment (bamader) [inline tests/linkage/test_postgres_mpi_connector.py:29]: 🤮

**Review by bamader — APPROVED** [review]
Based on the line 28 inclusion in the mpi connector test I can't in good conscience say this looks good, but great work on finding and fixing this issue!

**Review by emmastephenson — APPROVED** [review]
Changes LGTM, thanks for tracking down that bug!

**Review by m-goggins — APPROVED** [review]
Looks good!!

File: tests/linkage/test_postgres_mpi_connector.py (hunk 1 above)
Comment (DanPaseltiner) [reply tests/linkage/test_postgres_mpi_connector.py:29]: great then a simple `#noqa` it is :)

**Pull request merged**